src/status.rs

```diff
@@ -1603,36 +333,118 @@ impl Copy for StatusCode {}
 /// Formats the status code, *including* the canonical reason.
 ///
 /// ```rust
-/// # use hyper::status::StatusCode::{ImATeapot, Code123};
-/// # #[allow(unstable)]
-/// # fn main() {
-/// assert_eq!(&format!("{}", ImATeapot)[],
+/// # use hyper::status::{ImATeapot, UnregisteredStatusCode};
+/// assert_eq!(format!("{}", ImATeapot).as_slice(),
 ///            "418 I'm a teapot");
-/// assert_eq!(&format!("{}", Code123)[],
+/// assert_eq!(format!("{}", UnregisteredStatusCode(123)).as_slice(),
 ///            "123 <unknown status code>");
-/// # }
 /// ```
 ///
-/// If you wish to just include the number, cast to a u16 instead.
+/// If you wish to just include the number, cast to a `u16` instead:
+///
+/// ```rust
+/// # use std::num::ToPrimitive;
+/// # use hyper::status::{ImATeapot, UnregisteredStatusCode};
+/// assert_eq!(format!("{}", ImATeapot.to_u16().unwrap()).as_slice(), "418");
+/// assert_eq!(format!("{}", UnregisteredStatusCode(123).to_u16().unwrap()).as_slice(), "123");
+/// ```
 impl fmt::Display for StatusCode {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{} {}", *self as u16,
+        write!(f, "{} {}", self.to_u16().unwrap(),
                self.canonical_reason().unwrap_or("<unknown status code>"))
     }
 }
 
-// Specified manually because the codegen for derived is slow (at the time of writing on the machine
-// of writing, 1.2 seconds) and verbose (though the optimiser cuts it down to size).
+impl fmt::Debug for StatusCode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let tmp_s;
+        f.write_str(match *self {
+            UnregisteredStatusCode(ref code) => {
+                tmp_s = format!("StatusCode{}", code);
```

Comment:
Also, this would mean a code of like 999 would be `StatusCode999`. Is that actually desirable, or should it be `Unregistered(999)`?